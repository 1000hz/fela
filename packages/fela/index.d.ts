/* TODO: Add all Interfaces and Methods */
interface Renderer {
    renderRule(rule: Function, props?: any): string
}

export declare function createRenderer(config?:any): Renderer;
export declare function combineRules(...rules:any[]): Function;
export declare function enhance(...enhancers:any[]): Function;
export declare function render(renderer:any, mountNode:HTMLElement): void;
