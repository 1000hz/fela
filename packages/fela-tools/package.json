{
  "name": "fela-tools",
  "version": "4.1.2",
  "description": "Useful tools for working with Fela",
  "main": "index.js",
  "files": [
    "LICENSE",
    "README.md",
    "index.js",
    "StyleSheet.js",
    "mapValueToMediaQuery.js",
    "/dist"
  ],
  "repository": "https://github.com/rofrischmann/fela/",
  "keywords": [
    "fela",
    "fela-tool",
    "cssinjs"
  ],
  "author": "Robin Frischmann",
  "license": "MIT",
  "peerDependencies": {
    "fela": "4.1.2"
  }
}
