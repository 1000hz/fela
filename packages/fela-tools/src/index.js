export * from 'fela/lib/tools/index.js'
