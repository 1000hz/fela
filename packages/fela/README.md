# fela

NPM package to get access to Fela's API.<br>
Check out the [repository](https://github.com/rofrischmann/fela) for further information.
