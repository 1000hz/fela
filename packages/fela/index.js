module.exports = require('./lib/fela')
