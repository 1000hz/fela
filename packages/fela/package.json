{
  "name": "fela",
  "version": "5.2.0",
  "description": "Universal, Dynamic & High-Performance Styling in JavaScript",
  "typings": "index.d.ts",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "files": [
    "LICENSE",
    "README.md",
    "index.d.ts",
    "lib/**",
    "es/**"
  ],
  "repository": "https://github.com/rofrischmann/fela/",
  "keywords": [
    "fela",
    "dynamic styling",
    "stylesheet",
    "css",
    "styling",
    "cssinjs"
  ],
  "author": "Robin Frischmann",
  "license": "MIT",
  "dependencies": {
    "css-in-js-utils": "^2.0.0",
    "fela-utils": "^6.0.1",
    "recompose": "^0.25.0"
  }
}
