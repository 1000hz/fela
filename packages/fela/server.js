module.exports = require('lib/felaServer')
