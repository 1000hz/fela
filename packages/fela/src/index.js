/* @flow */
import createRenderer from './createRenderer'
import combineRules from './combineRules'
import enhance from './enhance'
import connectFactory from './bindings/connectFactory'
import createComponentFactory from './bindings/createComponentFactory'

export default {
  createRenderer,
  combineRules,
  enhance,
  connectFactory,
  createComponentFactory
}
