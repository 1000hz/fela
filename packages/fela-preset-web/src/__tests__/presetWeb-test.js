import { createRenderer } from 'fela'
import { renderToString } from 'fela-tools'

import webPreset, { createWebPreset } from '../index'

describe('preset-web-plugin', () => {
  it('should work without config', () => {
    const renderer = createRenderer({
      plugins: [...webPreset],
    })

    const rule = () => ({
      color: 'red',
      extend: {
        condition: true,
        style: {
          border: 'none',
        },
      },
    })

    renderer.renderRule(rule)
    // Tests that fela-plugin-extend is added to the plugins
    expect(renderToString(renderer)).toMatchSnapshot()
  })

  describe('configuration', () => {
    const renderer = createRenderer({
      plugins: [
        ...createWebPreset({
          unit: [
            'em',
            {
              margin: '%',
            },
          ],
        }),
      ],
    })

    it('should allow per plugin configuration', () => {
      renderer.renderRule(() => ({ width: 1 }))
      expect(renderToString(renderer)).toMatchSnapshot()
    })

    it('should pass all parameters to the plugins', () => {
      renderer.clear()
      renderer.renderRule(() => ({ margin: 1 }))
      expect(renderToString(renderer)).toMatchSnapshot()
    })
  })
})
