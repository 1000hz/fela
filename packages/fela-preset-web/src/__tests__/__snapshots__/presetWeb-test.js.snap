// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-web-plugin configuration should allow per plugin configuration 1`] = `".a{width:1em}"`;

exports[`preset-web-plugin configuration should pass all parameters to the plugins 1`] = `".a{margin:1%}"`;

exports[`preset-web-plugin should work without config 1`] = `".a{color:red}.b{border:none}"`;
